src/color/mfcamerareader.cpp

```diff
@@ -539,14 +539,165 @@ void CMFCameraReader::Stop()
     }
 }
 
+k4a_result_t CMFCameraReader::GetCameraControlCapabilities(const k4a_color_control_command_t command,
+                                                           color_control_cap_t &capabilities)
+{
+    HRESULT hr = S_OK;
+    bool supportAuto = false;
+    LONG minValue = 0;
+    LONG maxValue = 0;
+    ULONG stepValue = 0;
+    LONG defaultValue = 0;
+    k4a_color_control_mode_t defaultMode = K4A_COLOR_CONTROL_MODE_MANUAL;
+
+    switch (command)
+    {
+    case K4A_COLOR_CONTROL_EXPOSURE_TIME_ABSOLUTE:
+    {
+        hr = GetCameraControlCap(PROPSETID_VIDCAP_CAMERACONTROL,
+                                 KSPROPERTY_CAMERACONTROL_EXPOSURE,
+                                 &supportAuto,
+                                 &minValue,
+                                 &maxValue,
+                                 &stepValue,
+                                 &defaultValue);
+
+        // Convert KSProperty exposure time value to micro-second unit
+        minValue = (LONG)(exp2f((float)minValue) * 1000000.0f);
+        maxValue = (LONG)(exp2f((float)maxValue) * 1000000.0f);
+        stepValue = 1;
```

Comment:
>stepValue = 1; [](start = 8, length = 14)

is this true?  I thought stepping on this was nonlinear/exponential?  I don't know if we can even express that with this API, so I'm not entirely sure what to do with this one.  Probably at minimum we want some sort of comment if this isn't actually true

Reply:
I agree, this stepValue is actually not true for Media Foundation, but its "kind of true" in K4A API surface because it accept even its truncated into exponential value.
The other option for this API might be returning list of available values, but it's hard to make generic API for other controls and Linux version of API.
Any other better idea will be very appreciated.

Reply:
I see two possible approaches:

A)
I think setting it to 1 is fine since it'll still succeed, but we might want to stick a comment in here explaining that if you step by 1 it'll actually go up by a power of 2 so if someone gets confused and starts debugging, they'll understand why we're hard-setting it to 1 here.

B)
The only way I see to do a generic API here is to have an extra out parameter that's an enum of step types, something like 
```
enum K4A_COLOR_CONTROL_STEP_TYPE
{
    K4A_COLOR_CONTROL_STEP_TYPE_LINEAR,
    K4A_COLOR_CONTROL_STEP_TYPE_EXPONENTIAL
};
```

and then set everything except this one to linear.  This has the advantage of being more accurate, but adds complexity to every other color control that might not be worth it.

I don't have strong feelings on which approach to take here, but if we go with A) I do think adding a comment there explaining how it actually snaps to 1000000/[a power of 2] will probably save someone a fair amount of debugging at some point

---
In reply to: [281310570](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/pull/319#discussion_r281310570) [](ancestors = 281310570)

Reply:
I added some comment on it. Thanks.